src/data-types/char.ts

```diff
@@ -1,20 +1,25 @@
+import { DataType, ParameterData, Parameter } from '../data-type';
+
 const NULL = (1 << 16) - 1;
 
-module.exports = {
+const Char: { maximumLength: number } & DataType = {
   id: 0xAF,
   type: 'BIGCHAR',
   name: 'Char',
   hasCollation: true,
   dataLengthLength: 2,
   maximumLength: 8000,
 
-  declaration: function(parameter) {
+  declaration: function(parameter: Parameter) {
```

Comment:
I don't think you need to specify the type of `parameter` again.

```suggestion
  declaration: function(parameter) {
```